workflow/controller/operator.go

```diff
@@ -1989,6 +1992,23 @@ func (woc *wfOperationCtx) initializeCacheHitNode(nodeName string, resolvedTmpl
 	return node
 }
 
+func (woc *wfOperationCtx) finalizeTransformNode(nodeName string, data interface{}) *wfv1.NodeStatus {
```

Comment:
own file please

Reply:
There are a bunch of related functions that, if I were to move this to its own file, I'd want to move the rest with them. This PR or separate PR?